### ISSUE TITLE
Why is version 5 not compatible with node v4.x ?

### ISSUE THREAD

**Issue by schankam** [node-influx/node-influx#231]
Just a simple question here, but I was wondering why did you decide to develop the last version of node-influx based on node v6 (which is not a LTS version) instead of node v4.x ? Are you using any new features of v6 that they weren't any alternatives in v4 ?

In my case, I am getting an issue while trying to set username and password:

```
        [options.username, options.password] = parsed.auth.split(":");
        ^

ReferenceError: Invalid left-hand side in assignment
```

The problem here is that if we want to use the last version of your library, we need to upgrade to node v6, except that I am using other libraries working on node v4.x that will not work if I upgrade to node v6.

Maybe I am missing something here but I thought it would be much better to develop libraries and making them compatible with the current LTS version of node.

What do you think ?

**Comment by schankam** [comment]
Well, I just saw node v6 transition to LTS version... My bad then, I will try to see what I can do wit hmy other libraries.

**Comment by connor4312** [comment]
@schankam I should have Node 4 support published in the next ~15 minutes, as soon as Travis runs tests.

**Comment by connor4312** [comment]
While tests are running, I'll answer your question 😉 

Node 4 is an LTS release with support through 2018, it would make sense for us to continue supporting it. The reason you're getting the error is that, while Node 4 partially supports ES6, it doesn't have nearly full coverage of ES6 (particularly destructuring) like Node 6 does. Fortunately, since this project is TypeScript, I was able to switch to an ES5 code target with shimmed support for some ES6 features (like promises) in order to support Node 4.

**Comment by connor4312** [comment]
Node 4 support has been published in `5.0.0-alpha.4`. Please let me know if you hit any other snags 😄

**Comment by schankam** [comment]
Thanks for the efforts ! I've been using node 6 since, but good to know ;)
